sympy/polys/polytools.py

```diff
@@ -4404,13 +4404,20 @@ def degree(f, *gens, **args):
 
     """
     options.allowed_flags(args, ['gen', 'polys'])
-
+    f = sympify(f).as_expr()
     try:
-        F, opt = poly_from_expr(f, *gens, **args)
+        if f.is_Number:
```

Comment:
Yeah, but I think speed improvement would be negligible (I just passed it to `Poly` with gens when it's a number so I didn't have to re-code the logic, since `Poly` won't spend time collecting any terms for a number.) But I'll make another commit if it makes the code clearer